addons/mail/static/src/core/common/attachment_upload_service.js

```diff
@@ -46,49 +45,30 @@ export class AttachmentUploadService {
                 if (!this.uploadingAttachmentIds.has(tmpId)) {
                     return;
                 }
-                this.uploadingAttachmentIds.delete(tmpId);
-                this.abortByAttachmentId.delete(tmpId);
-                const { thread, composer } = this.targetsByTmpId.get(tmpId);
                 if (upload.xhr.status === 413) {
                     this.notificationService.add(_t("File too large"), { type: "danger" });
-                    this.targetsByTmpId.delete(tmpId);
+                    this._fileUploadBus.trigger("FILE_UPLOAD_ERROR", { upload });
```

Comment:
I know it was already there but I don't really see the point of using a bus rather than just calling the handler method directly.

Reply:
I found that both xhr.onerror and xhr.onabort events trigger the FILE_UPLOAD_ERROR event. I interpret FILE_UPLOAD_ERROR as indicating an upload failure. Therefore, regardless of the reason:

XHR error (xhr.onerror)
User abort (xhr.onabort)
Cloud server refusal (xhr.status !== 200)
The code should always call a single function to handle the cleanup process, which is the listener for FILE_UPLOAD_ERROR.

The statement this._fileUploadBus.trigger("FILE_UPLOAD_ERROR", { upload }); seems more like high-level business logic, while this.targetsByTmpId.delete(tmpId); is its implementation detail.

However, I will revert to using this.targetsByTmpId.delete(tmpId); since the code is still straightforward. I will leave the refactoring to your team for future improvements.